### PR TITLE
feat(kubernetes): add kubernetes section

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

Add Kubernetes section in manager 

WEBSITE-2317

### CODE REVIEW

File: client/app/kubernetes/kubernetes.constants.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+angular.module("managerApp")
+    .constant("KUBERNETES", {
+        // For now only one region is available
+        region: "GRA5",
```
Comment (FredericEspiau) [inline client/app/kubernetes/kubernetes.constants.js]: `regions`, think about the future 😄

File: client/app/kubernetes/kubernetes.controller.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        this.loading = true;
+        this.serviceName = this.$stateParams.serviceName;
+
+        this.getCluster();
+    }
+
+    getCluster () {
+        return this.Kubernetes.getKubernetesCluster(this.serviceName)
+            .then(cluster => { this.cluster = cluster; })
+            .catch(() => (this.cluster = { id: this.serviceName, name: this.serviceName }))
```
Comment (FredericEspiau) [inline client/app/kubernetes/kubernetes.controller.js]: Why ? What if the error is "server unavailable" ?

File: client/app/kubernetes/kubernetes.routes.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+angular.module("managerApp")
+    .config($stateProvider => {
+        $stateProvider
+            .state("paas.kube", {
+                url: "/kube/:serviceName",
+                templateUrl: "app/kubernetes/kubernetes.html",
+                controller: "KubernetesCtrl",
```
Comment (FredericEspiau) [inline client/app/kubernetes/kubernetes.routes.js]: You could have used a component 😄

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+angular.module("managerApp").controller("KubernetesNodesAddCtrl", class KubernetesNodesAddCtrl {
+
+    constructor ($scope, $q, $stateParams, $translate, $uibModalInstance, Kubernetes, projectId, CLOUD_FLAVORTYPE_CATEGORY) {
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: Alphabetical order

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    getFlavors () {
+        return this.Kubernetes.getFlavors(this.projectId)
+            .then(flavors => _.forEach(flavors, flavor => {
+                const family = _.find(this.CLOUD_FLAVORTYPE_CATEGORY, flavorType => _.includes(flavorType.types, flavor.type));
+                this.flavorsByFamily[family.id].push({ name: flavor.name, displayedName: this.Kubernetes.formatFlavor(flavor) });
+            }))
+            .finally(() => { this.loading = false; });
+
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: Whitespace

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        this.projectId = projectId;
+        this.CLOUD_FLAVORTYPE_CATEGORY = CLOUD_FLAVORTYPE_CATEGORY;
+    }
+
+    $onInit () {
+        this.loading = true;
+        this.serviceName = this.$stateParams.serviceName;
+
+        this.flavors = [];
+        this.flavorIds = _.map(this.CLOUD_FLAVORTYPE_CATEGORY, "id");
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: This looks complicated for what this is. Also, this could be done by a service.

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        this.flavorFamilies = _.map(this.flavorIds, flavorId => ({ id: flavorId, familyName: this.$translate.instant(`kube_nodes_add_flavor_family_${flavorId}`) }));
+        this.flavorsByFamily = _.zipObject(this.flavorIds, _.times(this.flavorFamilies.length, () => []));
+
+        this.getFlavors();
+    }
+
+    getFlavors () {
+        return this.Kubernetes.getFlavors(this.projectId)
+            .then(flavors => _.forEach(flavors, flavor => {
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: `foreach`+ `Array.prototype.push` === `map`

Once again, this looks complicated. Maybe we can think of different data type together, come to me if you have questions

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+                const family = _.find(this.CLOUD_FLAVORTYPE_CATEGORY, flavorType => _.includes(flavorType.types, flavor.type));
+                this.flavorsByFamily[family.id].push({ name: flavor.name, displayedName: this.Kubernetes.formatFlavor(flavor) });
+            }))
+            .finally(() => { this.loading = false; });
+
+    }
+
+    onFlavorFamilyChange (selectedFamily) {
+        this.flavor = null;
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: Why then ?

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        this.CloudMessage = CloudMessage;
+        this.Kubernetes = Kubernetes;
+        this.KUBERNETES = KUBERNETES;
+    }
+
+    $onInit () {
+        this.serviceName = this.$stateParams.serviceName;
+        this.loading = false;
+
+        this.loadMessages();
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: `Promise`s are meant to be chained :)

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+                }
+            }
+        }).result
+            .then(() => {
+                this.displaySuccessMessage("kube_nodes_delete_success");
+                this.Kubernetes.resetNodesCache();
+                this.getNodes();
+            })
+            .catch(error => {
+                if (error) {
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: You should display something even if the error is falsy

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+            backdrop: "static",
+            resolve: {
+                nodeId () {
+                    return nodeId;
+                }
+            }
+        }).result
+            .then(() => {
+                this.displaySuccessMessage("kube_nodes_delete_success");
+                this.Kubernetes.resetNodesCache();
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: I'm really confused about what's a `Promise` and what's not

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 12]
```diff
@@ -0,0 +1,40 @@
+<div data-ng-if="$ctrl.loading" class="text-center">
+    <oui-spinner data-size="m"></oui-spinner>
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: Sometimes `l`, sometimes `m`, is this justified ? You are supposed to use `m` if you are not loading a whole page

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+            </oui-column>
+            <oui-action-menu data-align="end" data-compact>
+                <oui-action-menu-item data-text="{{:: 'common_delete' | translate }}" data-on-click="$ctrl.confirmNodeDeletion($row.id)"></oui-action-menu-item>
+            </oui-action-menu>
+            <extra-top>
+                <oui-button data-ng-if="$ctrl.project"
+                            data-aria-label="{{:: 'kube_nodes_add' | translate }}"
+                            data-text="{{:: 'kube_nodes_add' | translate }}"
+                            data-on-click="$ctrl.openAddNodeForm($ctrl.project.project_id)"
+                            class="mb-1">
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: Why is this needed ?

File: client/app/kubernetes/service/kubernetes-service.html [hunk 14]
```diff
@@ -0,0 +1,81 @@
+<div data-ng-if="$ctrl.loaders.cluster || $ctrl.loaders.billing" class="text-center">
+    <oui-spinner data-size="m"></oui-spinner>
+</div>
+<div data-ng-if="!$ctrl.loaders.cluster && !$ctrl.loaders.billing">
+    <oui-message data-ng-if="$ctrl.displayError" data-type="error">
+        <span data-translate="kube_service_error"></span>
+    </oui-message>
+    <cui-message-container data-messages="$ctrl.messages"></cui-message-container>
+    <div>
```
Comment (FredericEspiau) [inline client/app/kubernetes/service/kubernetes-service.html]: Useless

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 15]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translations>
+    <translation id="kube_guides">Guides</translation>
```
Comment (FredericEspiau) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: Are those texts validated by `Content` ?

File: client/app/kubernetes/kubernetes.constants.js (hunk 1 above)
Comment (marie-j) [reply client/app/kubernetes/kubernetes.constants.js]: The region will be given by API in the future (it's the region of the cluster)

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js (hunk 7 above)
Comment (marie-j) [reply client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: The parsed array (`flavors`) is not the array filled (`flavorsByFamily[family]`) so I'm not sure `map` could do the trick

Yes we can try and think of something less complicated

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js (hunk 8 above)
Comment (marie-j) [reply client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: To prevent having a flavor still selected when the family changes

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js (hunk 10 above)
Comment (marie-j) [reply client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: `error` is undefined when the user dismisses the modal, I'm not sure an error message is needed then

Comment (marie-j) [reply client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: `error` is undefined when the user dismisses the modal, I'm not sure an error message is needed then

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+    <translation id="kube_service_billing_plan">Offre</translation>
+    <translation id="kube_service_billing_contact">Nic Billing</translation>
+    <translation id="kube_service_billing_renewal">Renouvellement</translation>
+
+    <translation id="kube_nodes">Noeuds</translation>
+    <translation id="kube_nodes_add">Ajouter un noeud</translation>
+    <translation id="kube_nodes_add_region">Durant la bêta, vos nœuds seront tous dans la région GRA5. Nous permettrons ultérieurement de choisir cette région.</translation>
+    <translation id="kube_nodes_add_success">Le noeud a bien été ajouté</translation>
+    <translation id="kube_nodes_add_flavor_family">Choisir un modèle d'instance</translation>
+    <translation id="kube_nodes_add_flavor_family_balanced">General purpose</translation>
```
Comment (cbourgois) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: EN text ? or legit ?

Comment (marie-j) [reply client/app/kubernetes/translations/Messages_fr_FR.xml]: This is the given name for french translations

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    <translation id="kube_service_error">Une erreur s'est produite lors du chargement des informations</translation>
+    <translation id="kube_service_description_information">Vous trouverez ci-dessous les informations de votre service Kubernetes, vous permettant d'y accéder à l'aide de l'</translation>
+    <translation id="kube_service_description_tool">outil kubetcl.</translation>
+    <translation id="kube_service_description_reset">Vous pouvez réinitialiser la configuration de votre cluster Kubernetes à tout moment.</translation>
+    <translation id="kube_service_offer_beta">Service managé Kubernetes Beta (gratuit)</translation>
+
+    <translation id="kube_service_file">Fichier kubeconfig</translation>
+    <translation id="kube_service_file_error">Une erreur est survenue lors du chargement du fichier de configuration</translation>
+    <translation id="kube_service_file_help">Le fichier kubeconfig contient les données permettant à l'outil en ligne de commande kubectl de communiquer avec l'API de votre cluster.</translation>
+    <translation id="kube_service_file_more_information">Plus d'information disponible ici</translation>
```
[the remark this answers is not in the document]
Comment (FredericEspiau) [reply client/app/kubernetes/translations/Messages_fr_FR.xml]: I've asked @marie-j to ask `Content` for their input

File: client/app/kubernetes/nodes/kubernetes-nodes.html (hunk 12 above)
Comment (marie-j) [reply client/app/kubernetes/nodes/kubernetes-nodes.html]: I have now put `m` sizes everywhere

File: client/app/kubernetes/kubernetes.service.js [hunk 18]
```diff
@@ -63,7 +63,7 @@ angular.module("managerApp").service("Kubernetes", class Kubernetes {
         return this.$translate.instant("kube_flavor", {
             name: flavor.name.toUpperCase(),
             cpuNumber: flavor.vcpus,
-            ramCapacity: flavor.ram,
+            ramCapacity: flavor.ram / 1000,
```
Comment (FredericEspiau) [inline client/app/kubernetes/kubernetes.service.js]: 😄

**Comment by ovh-cds** [comment]
CDS Report build-yarn#5962.0 ✘  
* Stage 1
  * deps ✘

**Comment by ovh-cds** [comment]
CDS Report build-manager#3.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-yarn#5967.0 ✘  
* Stage 1
  * deps ✘

File: package.json [hunk 19]
```diff
@@ -125,7 +125,7 @@
     "ovh-angular-tail-logs": "ovh-ux/ovh-angular-tail-logs#^1.1.2",
     "ovh-angular-toaster": "ovh-ux/ovh-angular-toaster#^0.8.0",
     "ovh-angular-user-pref": "ovh-ux/ovh-angular-user-pref#^0.3.1",
-    "ovh-api-services": "ovh-ux/ovh-api-services#^3.9.0",
+    "ovh-api-services": "ovh-ux/ovh-api-services#^3.15.1",
```
Comment (antleblanc) [inline package.json]: By upgrading the `ovh-api-services` dependency, `yarn.lock` file must be committed as well.

File: client/components/sidebar/products/kubernetes-sidebar.service.js [hunk 20]
```diff
@@ -0,0 +1,43 @@
+class KubernetesSidebar {
+  constructor($translate, OvhApiMe, SidebarMenu, URLS) {
```
Comment (antleblanc) [inline client/components/sidebar/products/kubernetes-sidebar.service.js]: You can remove `URLS` I guess.

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 21]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<translations>
+    <translation id="kube_guides">Guides</translation>
+    <translation id="kube_guides_kubernetes">Kubernetes</translation>
+
+    <translation id="kube_error">Une erreur s'est produite lors du chargement des informations: {{ message }}</translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: `informations: …` > `informations : …`

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+
+    <translation id="kube_service">Service</translation>
+    <translation id="kube_service_description_information">Vous trouverez ci-dessous les informations de votre service Kubernetes, vous permettant d'y accéder à l'aide de l'</translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: I suspect that you're doing that in order split your sentence with an anchor inside. But as translator point of view that's not convenient to deal with such a thing.

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+    <translation id="kube_service_billing">Facturation</translation>
+    <translation id="kube_service_billing_plan">Offre</translation>
+    <translation id="kube_service_billing_contact">Nic Billing</translation>
+    <translation id="kube_service_billing_renewal">Renouvellement</translation>
+
+    <translation id="kube_nodes">Nœuds</translation>
+    <translation id="kube_nodes_add">Ajouter un nœud</translation>
+    <translation id="kube_nodes_add_region">Durant la bêta, vos nœuds seront tous dans la région GRA5. Nous permettrons ultérieurement de choisir cette région.</translation>
+    <translation id="kube_nodes_add_success">Le nœud est en cours d'installation</translation>
+    <translation id="kube_nodes_add_flavor_error">Une erreur est survenue lors du chargement des modèles d'instance: {{ message }}</translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: `d'instance: …` > `d'instance : …`

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+    <translation id="kube_nodes_delete">Supprimer le nœud</translation>
+    <translation id="kube_nodes_delete_confirmation">Êtes-vous sûr de supprimer {{ node }} ?</translation>
+    <translation id="kube_nodes_delete_warning">Cette action est irréversible.</translation>
+    <translation id="kube_nodes_delete_enter">Entrez "DELETE" dans le champ ci-dessous pour confirmer</translation>
+    <translation id="kube_nodes_delete_success">Le nœud est en cours de suppression</translation>
+    <translation id="kube_nodes_delete_error">Une erreur est survenue lors de la suppression du nœud : {{ message }}</translation>
+
+    <translation id="kube_nodes_node">Nœud</translation>
+    <translation id="kube_nodes_error">Une erreur est survenue lors du chargement des nœuds</translation>
+    <translation id="kube_nodes_refresh">Recharger le tableau </translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: `…le tableau </translation>` > `…le tableau</translation>` (extra space)

File: client/app/kubernetes/translations/Messages_fr_FR.xml [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+    <translation id="kube_nodes_description_project_2">(nous permettrons de changer cela dans le futur). Ces instances vous seront facturées normalement dans votre projet public cloud.</translation>
+
+    <translation id="kube_flavor">{{ name }} ({{ cpuNumber }} CPU {{ ramCapacity }}Go RAM {{ diskCapacity }}Go Disque)</translation>
+
+    <translation id="kube_containers_and_services">Conteneurs et services</translation>
+    <translation id="kube_containers_coming_soon">Bientôt disponible</translation>
+    <translation id="kube_containers_description">Vous pourrez bientôt retrouver ici tous les conteneurs et services hébergés par votre cluster Kubernetes.</translation>
+    <translation id="kube_containers_description_dashboard">Nous vous invitons à utiliser l'outil en ligne de commande "kubectl" ou installer le Daskboard Kubernetes. Il s'agit d'une interface utilisateur web maintenue par la communauté Kubernetes, elle-même exposée en tant que service au sein de votre cluster. Suivez</translation>
```
Comment (antleblanc) [inline client/app/kubernetes/translations/Messages_fr_FR.xml]: Does `Suivez` could be move to the next translation key `kube_containers_description_dashboard_link`?
It seems more convenient from a translator point a view.

File: client/app/kubernetes/containers/kubernetes-containers.html [hunk 26]
```diff
@@ -0,0 +1,25 @@
+<div>
+    <h2 class="oui-header_2" data-translate="kube_containers_coming_soon"></h2>
+    <span data-translate="kube_containers_description"></span>
+    <div>
```
Comment (antleblanc) [inline client/app/kubernetes/containers/kubernetes-containers.html]: You can get rid of this `<div>` and for instance apply `d-block` helper class on your `<span>` element (or use a regular `<p>` element).

File: client/app/kubernetes/containers/kubernetes-containers.html [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+        <a class="oui-link oui-link_icon"
+           href="{{:: $ctrl.KUBERNETES.urls.kubernetesDashboard }}"
+           target="_blank"
+           rel="noopener">
+            <span data-translate="kube_containers_description_dashboard_link"></span>
+            <span class="oui-icon oui-icon-external_link" aria-hidden="true"></span>
+        </a>
+        <span data-translate="kube_containers_description_dashboard_2"></span>
+    </div>
+    <div>
```
Comment (antleblanc) [inline client/app/kubernetes/containers/kubernetes-containers.html]: Please refer to my previous comment. `<p>` element may help you.

File: client/app/kubernetes/containers/kubernetes-containers.routes.js [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+    $stateProvider
+      .state('paas.kube.containers', {
+        url: '/containers',
+        views: {
+          kubernetesView: {
+            component: 'kubernetesContainers',
+          },
+        },
+        resolve: {
+          serviceName: $transition$ => $transition$.params().serviceName,
```
Comment (antleblanc) [inline client/app/kubernetes/containers/kubernetes-containers.routes.js]: `$stateParams` can't be used?

File: client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+    this.loading = true;
+    this.serviceName = this.$stateParams.serviceName;
+
+    this.getFlavors();
+  }
+
+  getFlavors() {
+    return this.Kubernetes.getFlavors(this.projectId)
+      .then((flavors) => {
+        /**
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/add/kubernetes-nodes-add.controller.js]: indentation issue:)

File: client/app/kubernetes/nodes/kubernetes-nodes.html [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+                        'oui-status_warning' : $row.status === 'INSTALLING' || $row.status === 'DELETING',
+                        'oui-status_error' : $row.status === 'ERROR'
+                        }"
+                      data-translate="{{:: 'kube_service_cluster_status_' + $row.status }}">
+                </span>
+            </oui-column>
+            <oui-action-menu data-disabled="$row.status === 'INSTALLING' || $row.status === 'DELETING'"  data-align="end" data-compact>
+                <oui-action-menu-item data-text="{{:: 'common_delete' | translate }}" data-on-click="$ctrl.confirmNodeDeletion($row.id)"></oui-action-menu-item>
+            </oui-action-menu>
+            <extra-top>
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/kubernetes-nodes.html]: `<extra-top>` component encourage us to use `<oui-action-menu>` component.

example for the documentation:
```html
  <extra-top>
    <oui-action-menu text="Actions" aria-label="Server: actions" align="start">
      <oui-action-menu-item text="Add person" aria-label="Persons: add item"></oui-action-menu-item>
      <oui-action-menu-item text="Other action" aria-label="Persons: other action"></oui-action-menu-item>
    </oui-action-menu>
  </extra-top>
```
Can you tell me if that's fit your needs?

**Comment by marie-j** [comment]
@antleblanc changes done

File: client/app/kubernetes/nodes/kubernetes-nodes.html (hunk 30 above)
Comment (cbourgois) [reply client/app/kubernetes/nodes/kubernetes-nodes.html]: For 2 actions or more, use an action menu (guidelines).
